### PR TITLE
Pending and done tasks, also generate csv files

### PR DESCRIPTION
**Pull request by Mildred14**
**Description:** 
This branch has two different features. 

- First: Pending and done tasks

- Second: Generate csv files

**What implements/changes?:** 
Added classifications of tasks (pending and done)
An added feature of generating a CSV file

**How do you manually test this?: ** -

**Before - After screenshots:**
This is how it looks my interface 
image
image

### CODE REVIEW

File: db/migrate/20181018233104_add_user_id_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddUserIdToPosts < ActiveRecord::Migration[5.2]
+  def change
+    add_column :posts, :user_id, :integer
```
Comment (diegosaul402) [inline db/migrate/20181018233104_add_user_id_to_posts.rb:3]: The best solution here is explicitly adding the relation here and in your models users and posts.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user.id) 
+    @posts_pending_homework =  Post.where(done_homework: false, user_id: current_user.id)
```
Comment (diegosaul402) [inline app/controllers/posts_controller.rb]: When you make the relations correctly on your migrations and models, all of this kind of code 
will simplify to `@posts_done_homework = Post.where(done_homework: true, user: current_user)` you won't have to explicitly to specify that wich field keeps the relations between your models.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 <p id="notice"><%= notice %></p>
 
-<h1>Posts</h1>
+<p><%= link_to('Generate Report CSV', posts_path(format: 'csv')) %></p>
```
Comment (diegosaul402) [inline app/views/posts/index.html.erb]: Parenthesis for the `link_to` method is unnecessary, rails know that every sentence that follows are arguments.

**Review by diegosaul402 — CHANGES_REQUESTED** [review]
You're doing it great, but there is a big issue with the lack of relations for your models, you should fix this before continuing.

File: db/schema.rb [hunk 4]
```diff
@@ -10,13 +10,15 @@
...
 
-ActiveRecord::Schema.define(version: 2018_10_16_234314) do
+ActiveRecord::Schema.define(version: 2018_10_18_233104) do
 
   create_table "posts", force: :cascade do |t|
     t.string "title"
     t.text "body"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "user_id"
```
Comment (diegosaul402) [inline db/schema.rb:20]: your field can't be named `user_id`, just `user`

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -24,7 +33,7 @@ def edit
   # POST /posts
   # POST /posts.json
   def create
-    @post = Post.new(post_params)
+    @post = Post.new(post_params.merge(user_id: current_user.id))
```
Comment (diegosaul402) [inline app/controllers/posts_controller.rb:36]: if your relations are correct, you have to change your names `user_id` for `user` and `current_user.id` for `current_user`, this naming shouln't exist in any part of your code.

**Review by diegosaul402 — CHANGES_REQUESTED** [review]
You still need to do some changes in order to merge.

**Review by kikiondo — APPROVED** [review]

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user)
```
Comment (8geonirt) [inline app/controllers/posts_controller.rb:8]: Try to use ActiveRecord relations...
For example, if you have the User model linked to a Post, you can use something like this:
```
has_many :posts
```

And in your Post model
```
belongs_to :user
```

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user) 
+    @posts_pending_homework =  Post.where(done_homework: false, user_id: current_user)
```
Comment (8geonirt) [inline app/controllers/posts_controller.rb:9]: Same here...
Oh, what a coincidence... I saw the _done_homework_ variable in another project 😅

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user) 
+    @posts_pending_homework =  Post.where(done_homework: false, user_id: current_user) 
+
+    respond_to do |format|
+
```
Comment (8geonirt) [inline app/controllers/posts_controller.rb:12]: Remove this empty space

**Review by 8geonirt — CHANGES_REQUESTED** [review]
Some small details and suggestions 😅👍

**Review by 8geonirt — APPROVED** [review]

**Pull request merged**